pages/doc.js

```diff
@@ -55,35 +56,69 @@ export default class Documentation extends Component {
   }
 
   loadStateFromURL = () => {
-    const { pathname } = window.location
-    const sectionURL = pathname.split('/')[2] // match section from URL
-    const sectionIndex = sidebar.findIndex(
-      section => (section.slug || kebabCase(section.name)) === sectionURL
-    )
+    const { pathname } = window.location;
+    const sectionURL = pathname.split('/')[2];
+    const sectionIndex = sectionURL ? sidebar.findIndex(
+      section => kebabCase(section.name) === sectionURL
+    ) : 0;
     if (sectionIndex === -1) {
-      sectionURL
-        ? this.setState({ pageNotFound: true })
-        : this.onSectionSelect(0)
+      this.setState({ pageNotFound: true })
     } else {
-      const fileURL = pathname.split('/')[3] // match file from URL
-      const sectionFiles = flatten(sidebar[sectionIndex].files)
-      const fileIndex = sectionFiles.findIndex(
-        file => kebabCase(file.slice(0, -3)) === fileURL
-      )
-      if (fileIndex === -1) {
-        fileURL
-          ? this.setState({ pageNotFound: true })
-          : this.onSectionSelect(sectionIndex)
-      } else {
-        this.loadFile({
-          section: sectionIndex,
-          file: sectionFiles[fileIndex],
-          parseHeadings: true,
-          pageNotFound: false
-        })
+      const fileURL = pathname.split('/')[3]; // match file from URL
+      const section = sidebar[sectionIndex];
+      if(fileURL){
+        if(pathname.split('/')[4]){
```

Comment:
- the function is too long, too complicated too read and understand, it'll be very hard to support and modify it in the future
- url should be `docs/command-reference/remote/add` not `docs/command-reference/remote/remote-add`

Reply:
refactoring was done